api/analysis/dockerrun.go

```diff
@@ -248,3 +259,18 @@ func dockerPullImage(d *docker.Docker, image string) error {
 
 	return nil
 }
+
+func updateAndCheckContainerList(d *docker.Docker) error {
+	configAPI := context.GetAPIConfig()
+	maxContainersAllowed := configAPI.DockerHostsConfig.MaxContainersAllowed
+	listedContainers++
+	if listedContainers >= maxContainersAllowed {
+		err := d.DieContainers()
```

Comment:
I'm a little worried about how this function might perform. If `maxContainersAllowed` limit is reached, all running and not running containers will be stopped and removed, which, eventually, might stop ongoing huskyCI's analysis and crash huskyCI.

Reply:
Agree with @Krlier! To address this observation, I am suggesting to only list stopped containers (via a listStoppedContainers function) using the Filters parameter, which can be viewed at 1d8296e.

Reply:
Nice approach, everyone! 🙃